### PR TITLE
[WIP] [HotROD] Add optional fix to make driver.FindNearest call redis.GetDriver concurrently

### PR DESCRIPTION
**Pull request by tiffon**
## Which problem is this PR solving?

The HotROD demo can be optimized by calling `redis.GetDriver` concurrently.

## Short description of the changes

Adds a CLI flag (`-C`) to call `redis.GetDriver` concurrently.

-----

_This is my first foray into Go in some time, lmk if things look awry._

### CODE REVIEW

File: examples/hotrod/cmd/root.go [hunk 1]
```diff
@@ -64,6 +65,7 @@ func init() {
 	RootCmd.PersistentFlags().StringVarP(&metricsBackend, "metrics", "m", "expvar", "Metrics backend (expvar|prometheus)")
 	RootCmd.PersistentFlags().DurationVarP(&fixDBConnDelay, "fix-db-query-delay", "D", 300*time.Millisecond, "Average lagency of MySQL DB query")
 	RootCmd.PersistentFlags().BoolVarP(&fixDBConnDisableMutex, "fix-disable-db-conn-mutex", "M", false, "Disables the mutex guarding db connection")
+	RootCmd.PersistentFlags().BoolVarP(&fixGetDriversConcurrently, "fix-get-driver-concurrent", "C", false, "Enable concurrency in driver::findNearest")
```
Comment (yurishkuro) [inline examples/hotrod/cmd/root.go]: I don't think it should be a boolean, rather an integer (default 1), exactly like `fixRouteWorkerPoolSize`

File: examples/hotrod/services/driver/server.go [hunk 2]
```diff
@@ -72,31 +77,57 @@ func (s *Server) Run() error {
...
-// FindNearest implements Thrift interface TChanDriver
+// FindNearest implements Thrift interface TChanDriver. config.GetDriversConcurrently adds concurrency.
 func (s *Server) FindNearest(ctx thrift.Context, location string) ([]*driver.DriverLocation, error) {
 	s.logger.For(ctx).Info("Searching for nearby drivers", zap.String("location", location))
 	driverIDs := s.redis.FindDriverIDs(ctx, location)
-
 	retMe := make([]*driver.DriverLocation, len(driverIDs))
-	for i, driverID := range driverIDs {
-		var drv Driver
+	var topErr error
```
Comment (yurishkuro) [inline examples/hotrod/services/driver/server.go]: there seems to be a lot of unnecessary complexity here. See how `fixRouteWorkerPoolSize` is used, this should be exactly the same, using an executor pool.

Comment (tiffon) [reply examples/hotrod/services/driver/server.go]: Got it. Will change. Thanks.

File: examples/hotrod/cmd/root.go (hunk 1 above)
Comment (tiffon) [reply examples/hotrod/cmd/root.go]: SGTM. Will change.

**Comment by tiffon** [comment]
Closing as I haven't progressed this in some time.

**Pull request closed without merging**